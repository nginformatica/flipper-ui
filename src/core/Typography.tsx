import React from 'react'
import { DefaultProps } from './types'
import MuiTypography, { TypographyProps } from '@material-ui/core/Typography'

const Typography = ({
    children,
    margin,
    padding,
    style = {},
    variant = 'body2',
    ...otherProps
}: TypographyProps & DefaultProps) => (
    <MuiTypography
        {...otherProps}
        variant={variant}
        style={{ margin, padding, ...style }}>
        {children}
    </MuiTypography>
)

export default Typography
