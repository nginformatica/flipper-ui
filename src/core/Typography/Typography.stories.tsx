import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import Typography from '.'

export default {
    title: 'Typography',
    component: Typography
} as ComponentMeta<typeof Typography>

const Template: ComponentStory<typeof Typography> = args => (
    <Typography {...args}> {args.children} </Typography>
)

export const Default = Template.bind({})
Default.args = {
    children: 'This is a Typography text'
}

export const ErrorText = Template.bind({})
ErrorText.args = {
    color: 'error',
    children: 'Text with error color'
}

export const Primary = Template.bind({})
Primary.args = {
    color: 'primary',
    children: 'Text with primary color'
}

export const Secondary = Template.bind({})
Secondary.args = {
    color: 'secondary',
    children: 'Text with secondary color'
}

export const TextPrimary = Template.bind({})
TextPrimary.args = {
    color: 'textPrimary',
    children: 'Text with textPrimary color'
}

export const TextSecondary = Template.bind({})
TextSecondary.args = {
    color: 'textSecondary',
    children: 'Text with textSecondary color'
}

export const Button = Template.bind({})
Button.args = {
    variant: 'button',
    children: 'Button'
}

export const Caption = Template.bind({})
Caption.args = {
    variant: 'caption',
    children: 'Caption'
}

export const Body1 = Template.bind({})
Body1.args = {
    variant: 'body1',
    children: 'Body1'
}

export const Body2 = Template.bind({})
Body2.args = {
    variant: 'body2',
    children: 'Body2'
}

export const Subtitle1 = Template.bind({})
Subtitle1.args = {
    variant: 'subtitle1',
    children: 'Subtitle1'
}

export const Subtitle2 = Template.bind({})
Subtitle2.args = {
    variant: 'subtitle2',
    children: 'Subtitle2'
}

export const h1 = Template.bind({})
h1.args = {
    variant: 'h1',
    children: 'h1'
}

export const h2 = Template.bind({})
h2.args = {
    variant: 'h2',
    children: 'h2'
}

export const h3 = Template.bind({})
h3.args = {
    variant: 'h3',
    children: 'h3'
}

export const h4 = Template.bind({})
h4.args = {
    variant: 'h4',
    children: 'h4'
}

export const h5 = Template.bind({})
h5.args = {
    variant: 'h5',
    children: 'h5'
}

export const h6 = Template.bind({})
h6.args = {
    variant: 'h6',
    children: 'h6'
}
