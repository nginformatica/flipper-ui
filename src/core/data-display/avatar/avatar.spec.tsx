import React from 'react'
import { render, screen } from '@testing-library/react'
import Typography from '@/core/data-display/typography'
import { Avatar } from '.'

describe('Avatar', () => {
    it('should render first letter', () => {
        render(<Avatar>F</Avatar>)

        const avatar = screen.getByText('F')

        expect(avatar).toBeDefined()
    })

    it('should render primary color', () => {
        render(
            <Avatar primary data-testid='avatar-container'>
                F
            </Avatar>
        )

        const container = screen.getByTestId('avatar-container')

        const findPrimaryClass = Array.from(container.classList)
            .join(' ')
            .indexOf('makeStyles-primary-')

        expect(findPrimaryClass).toBeGreaterThan(-1)
    })

    it('should render with custom className', () => {
        render(
            <Avatar
                primary
                className={'custom-class'}
                data-testid='avatar-container'>
                F
            </Avatar>
        )

        const container = screen.getByTestId('avatar-container')

        expect(container.classList).toContain('custom-class')
    })

    it('should render nested children', () => {
        render(
            <Avatar>
                <div>
                    <Typography>icon</Typography>
                </div>
            </Avatar>
        )

        const avatar = screen.getByText('icon')

        expect(avatar).toBeDefined()
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Avatar primary className={'custom-class'}>
                F
            </Avatar>
        )

        expect(container).toMatchSnapshot()
    })
})
