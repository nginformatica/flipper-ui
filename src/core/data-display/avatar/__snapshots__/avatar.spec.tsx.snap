// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should match snapshot 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular custom-class makeStyles-primary-5 MuiAvatar-colorDefault"
  >
    F
  </div>
</div>
`;
